ESGF search fallback if index node is offline is still not working correctly

See the failing test [here](https://app.circleci.com/pipelines/github/ESMValGroup/ESMValCore/8133/workflows/54ea4469-bc86-441b-8af1-9b62f223f405/jobs/36870). Part of the stack trace:

```
tests/integration/esgf/test_search_download.py:145: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
esmvalcore/esgf/_search.py:335: in find_files
    return cached_search(**facets)
esmvalcore/esgf/_search.py:346: in cached_search
    files = esgf_search_files(esgf_facets)
esmvalcore/esgf/_search.py:152: in esgf_search_files
    files = ESGFFile._from_results(results, facets)
esmvalcore/esgf/_download.py:219: in _from_results
    results = sorted(results, key=same_file)
/opt/conda/envs/esmvaltool/lib/python3.10/_collections_abc.py:1038: in __iter__
    v = self[i]
/opt/conda/envs/esmvaltool/lib/python3.10/site-packages/pyesgf/search/results.py:47: in __getitem__
    batch = self.__get_batch(batch_i)
/opt/conda/envs/esmvaltool/lib/python3.10/site-packages/pyesgf/search/results.py:82: in __get_batch
    response = (self.context.connection
/opt/conda/envs/esmvaltool/lib/python3.10/site-packages/pyesgf/search/connection.py:159: in send_search
    response = self._send_query('search', full_query)
/opt/conda/envs/esmvaltool/lib/python3.10/site-packages/pyesgf/search/connection.py:203: in _send_query
    response = self.session.get(query_url, verify=self.verify,
/opt/conda/envs/esmvaltool/lib/python3.10/site-packages/requests/sessions.py:600: in get
    return self.request("GET", url, **kwargs)
/opt/conda/envs/esmvaltool/lib/python3.10/site-packages/requests_cache/session.py:115: in request
    return super().request(method, url, *args, **kwargs)
/opt/conda/envs/esmvaltool/lib/python3.10/site-packages/requests/sessions.py:587: in request
    resp = self.send(prep, **send_kwargs)
/opt/conda/envs/esmvaltool/lib/python3.10/site-packages/requests_cache/session.py:147: in send
    response = self._send_and_cache(request, actions, **kwargs)
/opt/conda/envs/esmvaltool/lib/python3.10/site-packages/requests_cache/session.py:189: in _send_and_cache
    response = super().send(request, **kwargs)
/opt/conda/envs/esmvaltool/lib/python3.10/site-packages/requests/sessions.py:701: in send
    r = adapter.send(request, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <requests.adapters.HTTPAdapter object at 0x7fc2d7800e80>
request = <PreparedRequest [GET]>, stream = False
timeout = Timeout(connect=120, read=120, total=None)
verify = '/opt/conda/envs/esmvaltool/lib/python3.10/site-packages/certifi/cacert.pem'
cert = None
proxies = OrderedDict([('no', '127.0.0.1,localhost,circleci-internal-outer-build-agent')])

    def send(
        self, request, stream=False, timeout=None, verify=True, cert=None, proxies=None
    ):
        """Sends PreparedRequest object. Returns Response object.
    
        :param request: The :class:`PreparedRequest <PreparedRequest>` being sent.
        :param stream: (optional) Whether to stream the request content.
        :param timeout: (optional) How long to wait for the server to send
            data before giving up, as a float, or a :ref:`(connect timeout,
            read timeout) <timeouts>` tuple.
        :type timeout: float or tuple or urllib3 Timeout object
        :param verify: (optional) Either a boolean, in which case it controls whether
            we verify the server's TLS certificate, or a string, in which case it
            must be a path to a CA bundle to use
        :param cert: (optional) Any user-provided SSL certificate to be trusted.
        :param proxies: (optional) The proxies dictionary to apply to the request.
        :rtype: requests.Response
        """
    
        try:
            conn = self.get_connection(request.url, proxies)
        except LocationValueError as e:
            raise InvalidURL(e, request=request)
    
        self.cert_verify(conn, request.url, verify, cert)
        url = self.request_url(request, proxies)
        self.add_headers(
            request,
            stream=stream,
            timeout=timeout,
            verify=verify,
            cert=cert,
            proxies=proxies,
        )
    
        chunked = not (request.body is None or "Content-Length" in request.headers)
    
        if isinstance(timeout, tuple):
            try:
                connect, read = timeout
                timeout = TimeoutSauce(connect=connect, read=read)
            except ValueError:
                raise ValueError(
                    f"Invalid timeout {timeout}. Pass a (connect, read) timeout tuple, "
                    f"or a single float to set both timeouts to the same value."
                )
        elif isinstance(timeout, TimeoutSauce):
            pass
        else:
            timeout = TimeoutSauce(connect=timeout, read=timeout)
    
        try:
            if not chunked:
                resp = conn.urlopen(
                    method=request.method,
                    url=url,
                    body=request.body,
                    headers=request.headers,
                    redirect=False,
                    assert_same_host=False,
                    preload_content=False,
                    decode_content=False,
                    retries=self.max_retries,
                    timeout=timeout,
                )
    
            # Send the request.
            else:
                if hasattr(conn, "proxy_pool"):
                    conn = conn.proxy_pool
    
                low_conn = conn._get_conn(timeout=DEFAULT_POOL_TIMEOUT)
    
                try:
                    skip_host = "Host" in request.headers
                    low_conn.putrequest(
                        request.method,
                        url,
                        skip_accept_encoding=True,
                        skip_host=skip_host,
                    )
    
                    for header, value in request.headers.items():
                        low_conn.putheader(header, value)
    
                    low_conn.endheaders()
    
                    for i in request.body:
                        low_conn.send(hex(len(i))[2:].encode("utf-8"))
                        low_conn.send(b"\r\n")
                        low_conn.send(i)
                        low_conn.send(b"\r\n")
                    low_conn.send(b"0\r\n\r\n")
    
                    # Receive the response from the server
                    r = low_conn.getresponse()
    
                    resp = HTTPResponse.from_httplib(
                        r,
                        pool=conn,
                        connection=low_conn,
                        preload_content=False,
                        decode_content=False,
                    )
                except Exception:
                    # If we hit any problems here, clean up the connection.
                    # Then, raise so that we can handle the actual exception.
                    low_conn.close()
                    raise
    
        except (ProtocolError, OSError) as err:
            raise ConnectionError(err, request=request)
    
        except MaxRetryError as e:
            if isinstance(e.reason, ConnectTimeoutError):
                # TODO: Remove this in 3.0.0: see #2811
                if not isinstance(e.reason, NewConnectionError):
>                   raise ConnectTimeout(e, request=request)
E                   requests.exceptions.ConnectTimeout: HTTPSConnectionPool(host='esgf.ceda.ac.uk', port=443): Max retries exceeded with url: /esg-search/search?format=application%2Fsolr%2Bjson&limit=500&distrib=true&offset=500&type=File&project=CMIP6&source_id=EC-Earth3&member_id=r1i1p1f1&experiment_id=historical&grid_label=gr&table_id=Amon&variable=tas (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x7fc2d7801210>, 'Connection to esgf.ceda.ac.uk timed out. (connect timeout=120)'))
```

--- a/esmvalcore/esgf/_search.py
+++ b/esmvalcore/esgf/_search.py
@@ -121,7 +121,7 @@ def _search_index_nodes(facets):
                 ignore_facet_check=True,
             )
             FIRST_ONLINE_INDEX_NODE = url
-            return results
+            return list(results)
         except (
             requests.exceptions.ConnectionError,
             requests.exceptions.HTTPError,

--- a/tests/unit/esgf/test_search.py
+++ b/tests/unit/esgf/test_search.py
@@ -263,7 +263,7 @@ def test_esgf_search_files(mocker):
 
 def test_esgf_search_uses_second_index_node(mocker):
     """Test that the second index node is used if the first is offline."""
-    search_result = mocker.sentinel.search_result
+    search_result = [mocker.sentinel.search_result]
     search_results = [
         requests.exceptions.ReadTimeout("Timeout error message"),
         search_result,

